Unable to access runtimes index on Windows Server 2019

**Describe the bug**
I'm not sure if there are any specific configuration steps that need to be taken for PyManager to work on Windows Server 2019, but I'm running into the following:
```
********************************************************************************
[ERROR] Unable to access runtimes index at https://www.python.org/ftp/python/index-windows.json: The parameter is incorrect.
[ERROR] Unable to find runtimes index at file:///C:/Program Files/PyManager/bundled/fallback-index.json
[ERROR] An error occurred. Please check any output above, or the log file, and try again.
Log file for this session: C:\Users\ADMINI~1\AppData\Local\Temp\2\python_install_20250515025034_1836.log
If you cannot resolve it yourself, please report the error with your log file at https://github.com/python/pymanager
To see all available commands, run 'py help'
********************************************************************************
[ERROR] Internal error 0x00000057. Please report to https://github.com/python/pymanager
```
It would seem that the download is failing *and* there was no fallback index bundled (probably just with the MSI install?).

**To Reproduce**
Steps to reproduce the behaviour:
1. Make a fresh install of Windows Server 2019 Standard with Desktop Experience (e.g. evaluation copy ISO download from Microsoft: https://www.microsoft.com/en-us/evalcenter/download-windows-server-2019)
2. Install MSI from https://www.python.org/ftp/python/pymanager/python-manager-25.0b6.msi
3. Run `py` in a fresh Windows PowerShell window.
4. See error.

**Expected behavior**
Probably no download issues 😄 
Considering that the error says "The parameter is incorrect", it seems like it might not be a configuration issue on my part but I don't actually know for sure, it's not a particularly descriptive error.

**Additional context**

Log file content:
```
>> Writing logs to C:\Users\ADMINI~1\AppData\Local\Temp\2\python_install_20250515025034_1836.log
>> Python installation manager 25.0b7
Copyright (c) Python Software Foundation. All Rights Reserved.

BEGIN install_command.execute: []
>  ********************************************************************************
Reading installs from C:\Users\Administrator\AppData\Local\Python
Found 0 installs
Calculating default install
Filtering installs by tag = 3.14
tag '3.14' matched 0 installs
default_platform '-64' matched 0 installs
Reusing unfiltered list
Searching https://www.python.org/ftp/python/index-windows.json
>> Searching for Python matching 3.14
Selecting package with tag=3.14 and platform=-64
Fetching: https://www.python.org/ftp/python/index-windows.json
winhttp_urlopen: https://www.python.org/ftp/python/index-windows.json
>> Failed to download using WinHTTP. Retrying with fallback method.
ERROR:
Traceback (most recent call last):
  File "manage\urlutils.py", line 361, in urlopen
  File "manage\urlutils.py", line 174, in _winhttp_urlopen
OSError: [WinError 87] The parameter is incorrect.

urllib download unavailable - using fallback
>> Failed to download using PowerShell. Retrying with fallback method.
ERROR:
Traceback (most recent call last):
  File "manage\urlutils.py", line 398, in urlopen
  File "manage\urlutils.py", line 272, in _powershell_urlopen
  File "manage\urlutils.py", line 286, in _powershell_urlretrieve
  File "manage\urlutils.py", line 89, in on_auth_request
  File "manage\urlutils.py", line 642, in on_auth
KeyError: 'https://www.python.org/ftp/python/index-windows.json'

!! Unable to access runtimes index at https://www.python.org/ftp/python/index-windows.json: The parameter is incorrect.
Capturing error in case fallbacks fail
Traceback (most recent call last):
  File "manage\install_command.py", line 719, in execute
  File "manage\install_command.py", line 394, in _find_one
  File "manage\install_command.py", line 66, in select_package
  File "manage\urlutils.py", line 660, in __next__
  File "manage\urlutils.py", line 408, in urlopen
  File "manage\urlutils.py", line 361, in urlopen
  File "manage\urlutils.py", line 174, in _winhttp_urlopen
OSError: [WinError 87] The parameter is incorrect.

Searching file:///C:/Program Files/PyManager/bundled/fallback-index.json
>> Searching for Python matching 3.14
Selecting package with tag=3.14 and platform=-64
Fetching: file:///C:/Program Files/PyManager/bundled/fallback-index.json
!! Unable to find runtimes index at file:///C:/Program Files/PyManager/bundled/fallback-index.json
Capturing error in case fallbacks fail
Traceback (most recent call last):
  File "manage\install_command.py", line 719, in execute
  File "manage\install_command.py", line 394, in _find_one
  File "manage\install_command.py", line 66, in select_package
  File "manage\urlutils.py", line 660, in __next__
  File "manage\urlutils.py", line 350, in urlopen
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Program Files\\PyManager\\bundled\\fallback-index.json'

!! An error occurred. Please check any output above, or the log file, and try again.
>  Log file for this session: C:\Users\ADMINI~1\AppData\Local\Temp\2\python_install_20250515025034_1836.log
>  If you cannot resolve it yourself, please report the error with your log file at https://github.com/python/pymanager
TRACEBACK:
Traceback (most recent call last):
  File "manage\install_command.py", line 734, in execute
  File "manage\install_command.py", line 719, in execute
  File "manage\install_command.py", line 394, in _find_one
  File "manage\install_command.py", line 66, in select_package
  File "manage\urlutils.py", line 660, in __next__
  File "manage\urlutils.py", line 408, in urlopen
  File "manage\urlutils.py", line 361, in urlopen
  File "manage\urlutils.py", line 174, in _winhttp_urlopen
OSError: [WinError 87] The parameter is incorrect.

>  To see all available commands, run 'py help'
>  ********************************************************************************
END install_command.execute
SILENCED ERROR
Traceback (most recent call last):
  File "manage\install_command.py", line 734, in execute
  File "manage\install_command.py", line 719, in execute
  File "manage\install_command.py", line 394, in _find_one
  File "manage\install_command.py", line 66, in select_package
  File "manage\urlutils.py", line 660, in __next__
  File "manage\urlutils.py", line 408, in urlopen
  File "manage\urlutils.py", line 361, in urlopen
  File "manage\urlutils.py", line 174, in _winhttp_urlopen
OSError: [WinError 87] The parameter is incorrect.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage\__init__.py", line 50, in main
  File "manage\commands.py", line 820, in execute
  File "manage\install_command.py", line 752, in execute
  File "manage\install_command.py", line 551, in _fatal_install_error
SystemExit: 87
```

--- a/src/_native/winhttp.cpp
+++ b/src/_native/winhttp.cpp
@@ -209,8 +209,21 @@ PyObject *winhttp_urlopen(PyObject *, PyObject *args, PyObject *kwargs) {
         WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY,
         WINHTTP_NO_PROXY_NAME,
         WINHTTP_NO_PROXY_BYPASS,
-        url_parts.nScheme == INTERNET_SCHEME_HTTPS ? WINHTTP_FLAG_SECURE_DEFAULTS : 0
+        url_parts.nScheme == INTERNET_SCHEME_HTTPS
+            ? WINHTTP_FLAG_SECURE_DEFAULTS & ~WINHTTP_FLAG_ASYNC
+            : 0
     );
+    if (!hSession && GetLastError() == ERROR_INVALID_PARAMETER) {
+        // WINHTTP_FLAG_SECURE_DEFAULTS is not supported on older OS, so we'll
+        // retry without it.
+        hSession = WinHttpOpen(
+            NULL,
+            WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY,
+            WINHTTP_NO_PROXY_NAME,
+            WINHTTP_NO_PROXY_BYPASS,
+            0
+        );
+    }
     CHECK_WINHTTP(hSession);
 
     hConnection = WinHttpConnect(

--- a/src/manage/config.py
+++ b/src/manage/config.py
@@ -36,6 +36,11 @@ def config_bool(v):
     return bool(v)
 
 
+def _is_valid_url(u):
+    from .urlutils import is_valid_url
+    return is_valid_url(u)
+
+
 def load_global_config(cfg, schema):
     try:
         from _native import package_get_root
@@ -184,7 +189,7 @@ def resolve_config(cfg, source, relative_to, key_so_far="", schema=None, error_u
             v = kind(v)
         except (TypeError, ValueError):
             raise InvalidConfigurationError(source, key_so_far + k, v)
-        if v and "path" in opts:
+        if v and "path" in opts and not _is_valid_url(v):
             # Paths from the config file are relative to the config file.
             # Paths from the environment are relative to the current working dir
             if not from_env:
@@ -196,8 +201,7 @@ def resolve_config(cfg, source, relative_to, key_so_far="", schema=None, error_u
                 v = v.as_uri()
             else:
                 v = str(v)
-            from .urlutils import is_valid_url
-            if not is_valid_url(v):
+            if not _is_valid_url(v):
                 raise InvalidConfigurationError(source, key_so_far + k, v)
         cfg[k] = v
 

--- a/src/manage/urlutils.py
+++ b/src/manage/urlutils.py
@@ -639,7 +639,10 @@ class IndexDownloader:
 
     def on_auth(self, url):
         # TODO: Try looking for parent paths from URL
-        return self._auth[url]
+        try:
+            return self._auth[url]
+        except LookupError:
+            return None
 
     def __next__(self):
         if not self._url:

--- a/src/pymanager/appxmanifest.xml
+++ b/src/pymanager/appxmanifest.xml
@@ -32,7 +32,7 @@
         <Resource Language="en-US" />
     </Resources>
     <Dependencies>
-        <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.17763.0" MaxVersionTested="10.0.26100.0" />
+        <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.19044.0" MaxVersionTested="10.0.26100.0" />
     </Dependencies>
     <Capabilities>
         <rescap:Capability Name="runFullTrust" />

